### PR TITLE
[QUIZ-603]: shared results page

### PR DESCRIPTION
**Pull request by gulevskydev**
For testing: 
Go to - https://codesandbox.io/s/quizzes-js-bundle-example-forked-cvhg3z
Insert URL: https://cvhg3z.csb.app/?attributes=red,blue,yellow&items=119011081
1) Should render results page
2) If clicked retake - should remove query params 'item' and 'attributes'  and start quiz again from first question

### CODE REVIEW

File: src/hooks/useQueryParams.ts
Comment (lws803) [inline src/hooks/useQueryParams.ts:13]: **suggestion**: I wonder if we should use `quizBasePath` prop like in my PR https://github.com/Constructor-io/constructorio-ui-quizzes/pull/145/files#diff-4ca8c4ece22292244e023432ad4ad692edfa1d8e0a64e44537935be42225251bR115 if supplied

Comment (lws803) [reply src/hooks/useQueryParams.ts:13]: but using window location href could be the fallback behavior though

File: src/hooks/useQuizState/useQuizApiState.ts
Comment (lws803) [inline src/hooks/useQuizState/useQuizApiState.ts:65]: **comment**: That came in handy :)

**Review by lws803 — CHANGES_REQUESTED** [review]
Looking good, really clean code. ✨  Tested it on the sandbox link you've shared, works like a charm!

Just one suggestion regarding considering the `quizBasePath` prop.

File: src/hooks/useQueryParams.ts
Comment (lws803) [inline src/hooks/useQueryParams.ts:14]: **issue**: Oh `quizBasePath` is not supplied as a query param here, it's supplied as a prop, see https://github.com/Constructor-io/constructorio-ui-quizzes/pull/145/files#diff-c54113cf61ec99691748a3890bfbeb00e10efb3f0a76f03a0fd9ec49072e410aR88 you might have to copy some of my changes over though

**Review by lws803 — CHANGES_REQUESTED** [review]

File: src/hooks/useQueryParams.ts
Comment (lws803) [reply src/hooks/useQueryParams.ts]: It will most likely be the same domain but just incase customers would like to pre-specify some query params, they can do that 👍 

> Because if we will try to update url with not the same domain we will get the error because of XSS

Ahh good shout, I don't think they'll specify any domain outside of their own 🤔

File: src/hooks/useQuizEvents/useQuizResetClick.ts
Comment (lws803) [inline src/hooks/useQuizEvents/useQuizResetClick.ts:22]: **suggestion**: I wonder if we should refactor this and pass an object as function parameter instead

File: src/hooks/useQueryParams.ts
Comment (lws803) [reply src/hooks/useQueryParams.ts]: As discussed, let's remove `quizBasePath` for now 👍 sorry for the trouble!

File: src/hooks/useQueryParams.ts
Comment (lws803) [inline src/hooks/useQueryParams.ts:5]: **comment**: Let's revert `quizBasePath` param as discussed 😅

**Review by lws803 — CHANGES_REQUESTED** [review]

**Review by lws803 — APPROVED** [review]
LGTM 👍 thanks for addressing the comments and sorry for the slight detour 😛 let's get a review from @mocca102 before merging though 👍

**Review by mocca102 — APPROVED** [review]
LGTM! The query parameters don't get removed on reset quiz when tested on storybook but might be smth only related to storybook. Also I was a bit concerned about rendering what's in the query params but React seems to be a good job escaping unwanted characters. Thanks for working on that!

**Comment by mocca102** [comment]
Also just noticed that on this url [](https://cvhg3z.csb.app/?attributes=red,blue,yellow&items=119011081)

Clicking retake quiz throws an error, Might be an edge case we need to handle when there's `/?` @v-i-s-h-n-u-ps @lws803 
Screenshot 2023-10-27 at 4 28 48 AM

**Comment by gulevskydev** [comment]
> https://cvhg3z.csb.app/?attributes=red,blue,yellow&items=119011081

Sorry, it was not the last version of compiled code in sandbox, now you can check the same logic using this URL: https://7q7w9x.csb.app/?attributes=red,blue,yellow&items=119011081

**Comment by gulevskydev** [comment]
> Also just noticed that on this url
> 
> Clicking retake quiz throws an error, Might be an edge case we need to handle when there's `/?` @v-i-s-h-n-u-ps @lws803 Screenshot 2023-10-27 at 4 28 48 AM

Sorry, it was not the last version of compiled code in sandbox, now you can check the same logic using this URL: https://7q7w9x.csb.app/?attributes=red,blue,yellow&items=119011081

**Pull request merged**
